### DISCUSSION TITLE
Importing from flickr... a guide

### DISCUSSION THREAD

**Discussion by brownphotographic** [immich-app/immich#12351] [General]
I'm new to Immich, and a 20 year flickr user. Id like to move my collection from flickr to immich, and was wondering how to export from flickr and import the images to immich.

The rough flow I am starting to experiment with:
1. You can request your data to be exported in flickr settings. It takes about a week, then flickr sends you an email. In the settings there then weill be a list of zip files: a set of json files with all the metadata, and a set of zips with 500 images per folder.
2. Use exiftool to scrape through the json and images, and apply exif to the images accordingly:
https://exiftool.org/forum/index.php?topic=11072.0
3. Upload to immich.

I will report back with progress.

**EDIT: published here...** https://github.com/brownphotographic/Flickr-Emmigration-Tool

**Comment by bitsondatadev** [comment]
Hey @brownphotographic,

I am also new to the community and I have a strange mashup of Apple and Google photos as a source. I had to jump around to a bunch of places to find all the details needed and wanted to make the onboarding migration process easier and help the core time improve docs and highlight features that could improve this experience.

My first step is creating this guide that I used: https://github.com/immich-app/immich/discussions/12334

Let me know if this helps and also I'd love to converge your steps into this article once they have solidified.

I'd love some feedback as well! :)

**Comment by brownphotographic** [comment]
Great! Thank you for the link. I will follow what you have and see where things need to differ for flickr. Good idea to stick it in one place.

**Comment by brownphotographic** [comment]
I'm still working on this... 

A useful link
https://exiftool.org/forum/index.php?topic=11072.msg74470#msg74470

**Reply by talllguy** [reply]
@brownphotographic Thanks for this topic. I wanted to know if you were successful in the endeavor. Did you get it working?

**Reply by satmandu** [reply]
Seconding this request for an update. I would also love a tested workflow for transferring from Flickr to Immich.

**Comment by brownphotographic** [comment]
Not yet! I have to write a script to get this done. I'm not a programmer
but have started the process. Will update when I'm done. It was supposed to
be a Xmas break project but my computer broke instead.

**Comment by brownphotographic** [comment]
OK, so part of the problem is my flickr account is HUGE. Trying to write code and debug this is very problematic. 

I spent some time just now to create a test album with some of my images with tags, albums, collections etc https://www.flickr.com/photos/202036170@N07/

Feel free to connect with me on the above account to add me as a friend and favorite photos, add comments etc. That would actually probably help to ensure I've got all the bases covered in the code.


(My main flickr account https://www.flickr.com/photos/brownphotographic/)

**Comment by brownphotographic** [comment]
OK, I am making progress!

I have created a python script and tested it on my small test account. It creates an output of folders (folders are album names), and embeds whatever metadata it can into the EXIF. The other fields not supported by EXIF I exported to XMP sidecar file.

Then you use the Immich CLI to import the folders. Works well... but I have small sample size so far.

One exception: Unfortunately it seems that Immich doesn't actually support XMP metadata yet. Even though this doc says XMP metadata is supported, it seems the cart was placed in front of the horse on that one.

Next steps:

1. I will test the code over the weekend, including parsing through my 20 year old Flickr library to see where it breaks.
2. I will adapt it to insert the additional flickr fields as a neatly formatted list into the EXIF Description. At least for now that makes the info visible.
3. I will publish the code on github, with an opensource software license (I'm a newbie to this, so hopefully github helps make that last part simple).
4. I will respond here when I've done that with a link to the repo. This will likely be buggy, and very much a prototype. It will need some other folks to help test it more.

Back to my day job!

https://github.com/immich-app/immich/discussions/15201
https://github.com/immich-app/immich/discussions/15200

**Reply by markstos** [reply]
Thanks for your work on this. I also have an ancient Flickr account and am a software dev. I could peer-review your code if you'd like.

**Reply by brownphotographic** [reply]
Hi all! Apologies for the slowness. 

I just published the first version (lets call this a clear alpha v0.1!) 

https://github.com/brownphotographic/Flickr-Emmigration-Tool

Of course I made this opensource. So go for it if you want to fork it, chip in with mods etc.

I had a heavy load of help from Claude.ai... which is awesome to help one code. I am not a SW developer (yeah so warning!) but a professional tech product manager that used to code way back. 

Hope this is helpful to you all. Note there are a TON of export options in the Readme file so have fun and read carefully to see what you'd like for exporting.

I will also upload a sample account export from my temporary account I created but do that to a different repo. Those images are not opensource. I will link that in the readme.

**Comment by brownphotographic** [comment]
Hi everyone! I just posted a new version:

    Added resume function
    Added log file summary
    Added functionality to process the highlights folder with public, private and friends and family

https://github.com/brownphotographic/Flickr-Emmigration-Tool/releases/tag/version

I suggest you watch the repo to get further updates. Please test and submit bugs and feature requests!

**Comment by brownphotographic** [comment]
I just published v0.3 with lots of bug fixes and new options.

https://github.com/brownphotographic/Flickr-Emmigration-Tool/releases/tag/v0.3
